src/Numerics/DeterminantOperators.h

```diff
@@ -232,6 +233,26 @@ inline typename MatA::value_type DetRatioByColumn(const MatA& Minv, const VecB&
   return simd::dot(Minv.cols(), Minv.data() + colchanged, Minv.cols(), newv.data(), 1);
 }
 
+/** determinant ratio with a column substitution
+ * @param Minv inverse matrix
+ * @param newv column vector
+ * @param colchanged column index to be replaced
+ * @return \f$ M^{new}/M\f$
+ */
+template<typename MatA, typename VecB>
+inline void mw_DetRatioByColumn(const int nw,
+                                int colchanged,
+                                const RefVector<MatA>& Minv_list,
+                                const RefVector<VecB>& newv_list,
+                                RefVector<typename MatA::value_type>& curRatio_list)
```

Comment:
Template on the value type instead of MatA and VecB. As a result, the type agreement between MatA and VecB can be enforced.
Replace curRatio_list RefVector with std::vector, there is no need of using reference_wrapper over scalars.

Reply:
Done